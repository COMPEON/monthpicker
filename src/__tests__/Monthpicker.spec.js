import React from 'react'
import { shallow, mount } from 'enzyme'
import Monthpicker from '..'

describe('<Monthpicker />', () => {
  it('renders its children', () => {
    const wrapper = shallow(
      <Monthpicker month={1} year={2010} onChange={jest.fn()}>Child</Monthpicker>
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('event handling: ', () => {
    it('opens the overlay when clicking on the child', () => {
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={1} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()

      child.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })

    it('closes the overlay when clicking anywhere outside the overlay', () => {
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={1} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()

      child.simulate('click')
      child.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })

    it('sets the focussed date to 1 month later when hitting ArrowRight', () => {
      const initialyFocussedMonth = 1
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={initialyFocussedMonth} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()
      child.simulate('click')

      wrapper.simulate('keyDown', { key: 'ArrowRight' })

      const focussedMonth = wrapper.findWhere(node => node.key() === 'Februar')
      expect(focussedMonth).toMatchSnapshot()
    })

    it('sets the focussed date to 1 month earlier when hitting ArrowLeft', () => {
      const initialyFocussedMonth = 2
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={initialyFocussedMonth} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()
      child.simulate('click')

      wrapper.simulate('keyDown', { key: 'ArrowLeft' })

      const focussedMonth = wrapper.findWhere(node => node.key() === 'Januar')
      expect(focussedMonth).toMatchSnapshot()
    })

    it('sets the focussed date to 3 month later when hitting ArrowDown', () => {
      const initialyFocussedMonth = 1
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={initialyFocussedMonth} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()
      child.simulate('click')

      wrapper.simulate('keyDown', { key: 'ArrowDown' })

      const focussedMonth = wrapper.findWhere(node => node.key() === 'April')
      expect(focussedMonth).toMatchSnapshot()
    })

    it('sets the focussed date to 3 month earlier when hitting ArrowUp', () => {
      const initialyFocussedMonth = 4
      const wrapper = mount(
        <Monthpicker initialYear={2010} month={initialyFocussedMonth} year={2010} onChange={jest.fn()}>
          <div className='child'>Child</div>
        </Monthpicker>
      )

      const child = wrapper.find('.child').first()
      child.simulate('click')

      wrapper.simulate('keyDown', { key: 'ArrowUp' })

      const focussedMonth = wrapper.findWhere(node => node.key() === 'Januar')
      expect(focussedMonth).toMatchSnapshot()
    })
  })

  it('calls onChange with the selected month and year', () => {
    const handleChange = jest.fn()
    const wrapper = mount(
      <Monthpicker month={1} year={2018} onChange={handleChange}>
        <div className='child'>Child</div>
      </Monthpicker>
    )

    const child = wrapper.find('.child').first()
    child.simulate('click')

    wrapper.findWhere(node => node.key() === 'Mai').simulate('click')

    expect(handleChange).toHaveBeenCalledWith({ month: 5, year: 2018 }, expect.anything())
  })

  it('calls onChange with the given format', () => {
    const handleChange = jest.fn()
    const targetFormat = 'MM.YYYY'

    const wrapper = mount(
      <Monthpicker format={targetFormat} month={1} year={2018} onChange={handleChange}>
        <div className='child'>Child</div>
      </Monthpicker>
    )

    const child = wrapper.find('.child').first()
    child.simulate('click')

    wrapper.findWhere(node => node.key() === 'Mai').simulate('click')

    expect(handleChange).toHaveBeenCalledWith('05.2018', expect.anything())
  })

  it('allows specifiyng the primary and secondary colors', () => {
    const wrapper = mount(
      <Monthpicker
        primaryColor='green'
        secondaryColor='red'
        month={1}
        year={2018}
        onChange={jest.fn()}
      >
        <div className='child'>Child</div>
      </Monthpicker>
    )

    const child = wrapper.find('.child').first()
    child.simulate('click')

    expect(wrapper).toMatchSnapshot()
  })
})
