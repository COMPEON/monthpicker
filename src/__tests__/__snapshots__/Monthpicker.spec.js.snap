// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Monthpicker /> allows specifiyng the primary and secondary colors 1`] = `
.c1 {
  margin-top: 8px;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  -webkit-animation: dXhcnZ .2s;
  animation: dXhcnZ .2s;
  position: absolute;
  box-shadow: 0 1px 3px #d3d3d380,0 1px 3px #d3d3d380;
  border: 1px solid green;
  border-radius: 4px;
  z-index: 10;
}

.c8 {
  border: 1px solid #d3d3d37a;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: green;
  color: red;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c8:focus {
  outline: none;
}

.c9 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: red;
  color: green;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c9:hover {
  background-color: #d3d3d330;
}

.c9:focus {
  outline: none;
}

.c10 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: red;
  color: green;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
  border-bottom-left-radius: 4px;
}

.c10:hover {
  background-color: #d3d3d330;
}

.c10:focus {
  outline: none;
}

.c11 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: red;
  color: green;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
  border-bottom-right-radius: 4px;
}

.c11:hover {
  background-color: #d3d3d330;
}

.c11:focus {
  outline: none;
}

.c5 {
  cursor: pointer;
  fill: red;
}

.c3 {
  cursor: pointer;
  fill: white;
  fill: red;
}

.c2 {
  padding: 16px 8px;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: green;
  color: red;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 * {
  box-sizing: border-box;
}

.c0:focus {
  outline: none;
}

.c7 {
  background-color: white;
  border-radius: 4px;
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 {
  height: 8px;
  background-color: white;
  width: 100%;
}

<MonthPicker
  hoverColor="#d3d3d330"
  initialYear={2018}
  locale="de"
  month={1}
  onChange={[MockFunction]}
  primaryColor="green"
  secondaryColor="red"
  year={2018}
>
  <styled.div
    innerRef={[Function]}
    onKeyDown={[Function]}
    tabIndex={-1}
  >
    <div
      className="c0"
      onKeyDown={[Function]}
      tabIndex={-1}
    >
      <div
        onClick={[Function]}
      >
        <div
          className="child"
        >
          Child
        </div>
      </div>
      <styled.div
        hoverColor="#d3d3d330"
        primaryColor="green"
        secondaryColor="red"
      >
        <div
          className="c1"
        >
          <styled.div
            hoverColor="#d3d3d330"
            primaryColor="green"
            secondaryColor="red"
          >
            <div
              className="c2"
            >
              <Component
                hoverColor="#d3d3d330"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
              >
                <styled.svg
                  height="24"
                  hoverColor="#d3d3d330"
                  onClick={[Function]}
                  primaryColor="green"
                  secondaryColor="red"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    className="c3"
                    height="24"
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                    <path
                      d="M0-.5h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </styled.svg>
              </Component>
              <styled.span>
                <span
                  className="c4"
                >
                  2018
                </span>
              </styled.span>
              <Component
                hoverColor="#d3d3d330"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
              >
                <styled.svg
                  height="24"
                  hoverColor="#d3d3d330"
                  onClick={[Function]}
                  primaryColor="green"
                  secondaryColor="red"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    className="c5"
                    height="24"
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                    <path
                      d="M0-.25h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </styled.svg>
              </Component>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c6"
            />
          </styled.div>
          <styled.div
            hoverColor="#d3d3d330"
            primaryColor="green"
            secondaryColor="red"
          >
            <div
              className="c7"
            >
              <styled.div
                focussed={true}
                hoverColor="#d3d3d330"
                index={0}
                key="M01"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={true}
              >
                <div
                  className="c8"
                  onClick={[Function]}
                  selected={true}
                >
                  M01
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={1}
                key="M02"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M02
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={2}
                key="M03"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M03
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={3}
                key="M04"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M04
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={4}
                key="M05"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M05
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={5}
                key="M06"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M06
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={6}
                key="M07"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M07
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={7}
                key="M08"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M08
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={8}
                key="M09"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M09
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={9}
                key="M10"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c10"
                  onClick={[Function]}
                  selected={false}
                >
                  M10
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={10}
                key="M11"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M11
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={11}
                key="M12"
                onClick={[Function]}
                primaryColor="green"
                secondaryColor="red"
                selected={false}
              >
                <div
                  className="c11"
                  onClick={[Function]}
                  selected={false}
                >
                  M12
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</MonthPicker>
`;

exports[`<Monthpicker /> event handling:  closes the overlay when clicking anywhere outside the overlay 1`] = `
.c0 * {
  box-sizing: border-box;
}

.c0:focus {
  outline: none;
}

<MonthPicker
  hoverColor="#d3d3d330"
  initialYear={2010}
  locale="de"
  month={1}
  onChange={[MockFunction]}
  primaryColor="#27718c"
  secondaryColor="white"
  year={2010}
>
  <styled.div
    innerRef={[Function]}
    onKeyDown={[Function]}
    tabIndex={-1}
  >
    <div
      className="c0"
      onKeyDown={[Function]}
      tabIndex={-1}
    >
      <div
        onClick={[Function]}
      >
        <div
          className="child"
        >
          Child
        </div>
      </div>
    </div>
  </styled.div>
</MonthPicker>
`;

exports[`<Monthpicker /> event handling:  opens the overlay when clicking on the child 1`] = `
.c1 {
  margin-top: 8px;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  -webkit-animation: dXhcnZ .2s;
  animation: dXhcnZ .2s;
  position: absolute;
  box-shadow: 0 1px 3px #d3d3d380,0 1px 3px #d3d3d380;
  border: 1px solid #27718c;
  border-radius: 4px;
  z-index: 10;
}

.c8 {
  border: 1px solid #d3d3d37a;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: #27718c;
  color: white;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c8:focus {
  outline: none;
}

.c9 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: white;
  color: #27718c;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c9:hover {
  background-color: #d3d3d330;
}

.c9:focus {
  outline: none;
}

.c10 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: white;
  color: #27718c;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
  border-bottom-left-radius: 4px;
}

.c10:hover {
  background-color: #d3d3d330;
}

.c10:focus {
  outline: none;
}

.c11 {
  cursor: pointer;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: white;
  color: #27718c;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
  border-bottom-right-radius: 4px;
}

.c11:hover {
  background-color: #d3d3d330;
}

.c11:focus {
  outline: none;
}

.c5 {
  cursor: pointer;
  fill: white;
}

.c3 {
  cursor: pointer;
  fill: white;
  fill: white;
}

.c2 {
  padding: 16px 8px;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #27718c;
  color: white;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 * {
  box-sizing: border-box;
}

.c0:focus {
  outline: none;
}

.c7 {
  background-color: white;
  border-radius: 4px;
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 {
  height: 8px;
  background-color: white;
  width: 100%;
}

<MonthPicker
  hoverColor="#d3d3d330"
  initialYear={2010}
  locale="de"
  month={1}
  onChange={[MockFunction]}
  primaryColor="#27718c"
  secondaryColor="white"
  year={2010}
>
  <styled.div
    innerRef={[Function]}
    onKeyDown={[Function]}
    tabIndex={-1}
  >
    <div
      className="c0"
      onKeyDown={[Function]}
      tabIndex={-1}
    >
      <div
        onClick={[Function]}
      >
        <div
          className="child"
        >
          Child
        </div>
      </div>
      <styled.div
        hoverColor="#d3d3d330"
        primaryColor="#27718c"
        secondaryColor="white"
      >
        <div
          className="c1"
        >
          <styled.div
            hoverColor="#d3d3d330"
            primaryColor="#27718c"
            secondaryColor="white"
          >
            <div
              className="c2"
            >
              <Component
                hoverColor="#d3d3d330"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
              >
                <styled.svg
                  height="24"
                  hoverColor="#d3d3d330"
                  onClick={[Function]}
                  primaryColor="#27718c"
                  secondaryColor="white"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    className="c3"
                    height="24"
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                    <path
                      d="M0-.5h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </styled.svg>
              </Component>
              <styled.span>
                <span
                  className="c4"
                >
                  2010
                </span>
              </styled.span>
              <Component
                hoverColor="#d3d3d330"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
              >
                <styled.svg
                  height="24"
                  hoverColor="#d3d3d330"
                  onClick={[Function]}
                  primaryColor="#27718c"
                  secondaryColor="white"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    className="c5"
                    height="24"
                    onClick={[Function]}
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                    <path
                      d="M0-.25h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </styled.svg>
              </Component>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c6"
            />
          </styled.div>
          <styled.div
            hoverColor="#d3d3d330"
            primaryColor="#27718c"
            secondaryColor="white"
          >
            <div
              className="c7"
            >
              <styled.div
                focussed={true}
                hoverColor="#d3d3d330"
                index={0}
                key="M01"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={true}
              >
                <div
                  className="c8"
                  onClick={[Function]}
                  selected={true}
                >
                  M01
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={1}
                key="M02"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M02
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={2}
                key="M03"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M03
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={3}
                key="M04"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M04
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={4}
                key="M05"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M05
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={5}
                key="M06"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M06
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={6}
                key="M07"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M07
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={7}
                key="M08"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M08
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={8}
                key="M09"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M09
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={9}
                key="M10"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c10"
                  onClick={[Function]}
                  selected={false}
                >
                  M10
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={10}
                key="M11"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c9"
                  onClick={[Function]}
                  selected={false}
                >
                  M11
                </div>
              </styled.div>
              <styled.div
                focussed={false}
                hoverColor="#d3d3d330"
                index={11}
                key="M12"
                onClick={[Function]}
                primaryColor="#27718c"
                secondaryColor="white"
                selected={false}
              >
                <div
                  className="c11"
                  onClick={[Function]}
                  selected={false}
                >
                  M12
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</MonthPicker>
`;

exports[`<Monthpicker /> event handling:  sets the focussed date to 1 month earlier when hitting ArrowLeft 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #27718c;
  background-color: #d3d3d330;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: white;
  color: #27718c;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c0:hover {
  background-color: #d3d3d330;
}

.c0:focus {
  outline: none;
}

.c0:nth-child(3n) {
  border-right: 1px solid white;
}

.c0:nth-child(3n+1) {
  border-left: 1px solid white;
}

.c0:nth-last-child(-n+3) {
  border-bottom: 1px solid white;
}

<styled.div
  focussed={true}
  hoverColor="#d3d3d330"
  index={0}
  key="M01"
  onClick={[Function]}
  primaryColor="#27718c"
  secondaryColor="white"
  selected={false}
>
  <div
    className="c0"
    onClick={[Function]}
    selected={false}
  >
    M01
  </div>
</styled.div>
`;

exports[`<Monthpicker /> event handling:  sets the focussed date to 1 month later when hitting ArrowRight 1`] = `
.c0 {
  cursor: pointer;
  border: 1px solid #27718c;
  background-color: #d3d3d330;
  padding: 12px 0;
  -webkit-transition: background-color .1s,color .1s;
  transition: background-color .1s,color .1s;
  font-size: 14px;
  background-color: white;
  color: #27718c;
  height: 42px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 33.33%;
  -ms-flex: 0 0 33.33%;
  flex: 0 0 33.33%;
}

.c0:hover {
  background-color: #d3d3d330;
}

.c0:focus {
  outline: none;
}

.c0:nth-child(3n) {
  border-right: 1px solid white;
}

.c0:nth-child(3n+1) {
  border-left: 1px solid white;
}

.c0:nth-last-child(-n+3) {
  border-bottom: 1px solid white;
}

<styled.div
  focussed={true}
  hoverColor="#d3d3d330"
  index={1}
  key="M02"
  onClick={[Function]}
  primaryColor="#27718c"
  secondaryColor="white"
  selected={false}
>
  <div
    className="c0"
    onClick={[Function]}
    selected={false}
  >
    M02
  </div>
</styled.div>
`;

exports[`<Monthpicker /> event handling:  sets the focussed date to 3 month earlier when hitting ArrowUp 1`] = `null`;

exports[`<Monthpicker /> event handling:  sets the focussed date to 3 month later when hitting ArrowDown 1`] = `null`;

exports[`<Monthpicker /> renders its children 1`] = `
<styled.div
  innerRef={[Function]}
  onKeyDown={[Function]}
  tabIndex={-1}
>
  <div
    onClick={[Function]}
  >
    Child
  </div>
</styled.div>
`;
